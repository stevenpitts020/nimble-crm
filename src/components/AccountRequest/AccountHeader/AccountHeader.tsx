import React from 'react'
import { useHistory } from 'react-router-dom'
import { Typography, Row, Button, Col } from 'antd'
import { Stack, ArrowLeftIcon } from '../../index'
import { AccountRequestInfo } from './AccountRequestInfo'
import { ModalBSARiskResults } from '../ModalBSARiskResults/ModalBSARiskResults'

interface IAccountHeader {
  title?: string
  accountNumber: string
  accountRequestId: string
  branchName?: string
  contract?: {
    uri: string
  }
  bsa?: {
    score: number
    risk: string
  }
}

export const AccountHeader: React.FC<IAccountHeader> = (
  props: IAccountHeader
) => {
  const { accountNumber, accountRequestId, contract, title, bsa } = props

  const { Title, Text } = Typography
  const history = useHistory()

  const handleGoBack = () => {
    history.goBack()
  }

  return (
    <Row
      justify="space-between"
      align="middle"
      className="account-request-header u-margin-bottom-xl"
      data-testid="ni-account-request-header"
    >
      <Col xs={24} sm={8} lg={8}>
        <Stack direction="horizontal">
          <Button
            type="link"
            shape="circle"
            onClick={handleGoBack}
            data-testid="goback-button"
          >
            <ArrowLeftIcon className="u-margin-left-xs u-margin-right-md ni-color-black" />
          </Button>
          <Stack direction="vertical">
            <Text className="ni-color-silver" strong={true}>
              Account #{accountNumber}{' '}
              {props.branchName ? `- ${props.branchName}` : ''}
            </Text>
            <Title level={2} className="no-margin">
              {title}
            </Title>
            <ModalBSARiskResults
              accountRequestId={accountRequestId}
              bsa={
                bsa || {
                  score: 0,
                  risk: 'Pending',
                }
              }
            />
          </Stack>
        </Stack>
      </Col>
      <Col xs={24} sm={16} lg={16}>
        {contract && <AccountRequestInfo uri={contract.uri} />}
      </Col>
    </Row>
  )
}
