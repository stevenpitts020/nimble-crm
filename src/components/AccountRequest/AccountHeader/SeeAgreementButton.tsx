import React from 'react'
import { Button } from 'antd'
import {
  CheckCircleWhiteIcon,
  AlertTriangleIconWhite,
} from '../../Common/Icon/Icon'

interface IAgreementButton {
  uri: string
  accountRequestStatus: string
}
const composeButtonClassName = (completed: boolean, declined: boolean) => {
  const className = ['agreement-button']
  if (completed) {
    className.push('completed')
  }
  if (declined) {
    className.push('declined')
  }
  return className.join(' ')
}

export const SeeAgreementButton: React.FC<IAgreementButton> = ({
  uri,
  accountRequestStatus,
}) => {
  const completed =
    accountRequestStatus === 'COMPLETED' ||
    accountRequestStatus === 'SIGNED' ||
    accountRequestStatus === 'APPROVED'

  const declined = accountRequestStatus === 'DECLINED'

  return (
    <a
      href={uri}
      target="_blank"
      rel="noreferrer noopener"
      data-testid="agreement-button"
      data-status={accountRequestStatus}
    >
      <Button
        shape="round"
        className={composeButtonClassName(completed, declined)}
      >
        {completed ? (
          <CheckCircleWhiteIcon data-testid="signed" />
        ) : (
          <AlertTriangleIconWhite data-testid="pending" />
        )}
        See Agreement
      </Button>
    </a>
  )
}
