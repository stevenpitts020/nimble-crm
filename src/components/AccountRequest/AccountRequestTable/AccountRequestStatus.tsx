import React from 'react'
import { IAccountRequest } from '../../../store'
import FormatHelper from '../../../utils/FormatHelper'

interface IAccountRequestStatus {
  accountRequest: IAccountRequest
  isStatusClickable?: (accountRequest: IAccountRequest) => boolean
  onStatusClick?: (accountRequest: IAccountRequest) => void
}

export const AccountRequestStatus: React.FC<IAccountRequestStatus> = (
  props: IAccountRequestStatus
) => {
  const acctReq = props.accountRequest
  const status = acctReq.status
  const mappedStatus = FormatHelper.getAccountRequestStatus(status)
  const clickable = props.isStatusClickable
    ? props.isStatusClickable(acctReq)
    : false

  const onClick = () => {
    if (clickable && props.onStatusClick) props.onStatusClick(acctReq)
  }

  const style = {} as any
  if (clickable) style.cursor = 'pointer'

  return (
    <div className="account-request-status">
      {status === 'INCOMPLETE' && (
        <button onClick={onClick} className="incomplete" style={style}>
          {mappedStatus}
        </button>
      )}
      {status === 'PENDING' && (
        <button onClick={onClick} className="pending" style={style}>
          {mappedStatus}
        </button>
      )}
      {status === 'SIGNED' && (
        <button onClick={onClick} className="signed" style={style}>
          {mappedStatus}
        </button>
      )}
      {status === 'APPROVED' && (
        <button onClick={onClick} className="approved" style={style}>
          {mappedStatus}
        </button>
      )}
      {status === 'DECLINED' && (
        <button onClick={onClick} className="declined" style={style}>
          {mappedStatus}
        </button>
      )}
    </div>
  )
}
