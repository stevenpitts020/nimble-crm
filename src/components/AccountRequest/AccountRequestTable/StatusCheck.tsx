import React from 'react'
import { Tooltip } from 'antd'
import { ValidationIcon } from '../../Signer/SignerVerification/SignerVerificationStatus'

interface IStatusCheck {
  title: string
  verificationCheck: 'VALID' | 'INVALID' | 'PENDING' | 'NOMATCH' | 'MATCH'
}

export const StatusCheck: React.FC<IStatusCheck> = ({
  title,
  verificationCheck,
}) => {
  return (
    <Tooltip title={title}>
      <div className="status-check-item">
        <span aria-hidden="true" hidden={true}>
          {title}
        </span>

        <ValidationIcon check={verificationCheck} />
      </div>
    </Tooltip>
  )
}
