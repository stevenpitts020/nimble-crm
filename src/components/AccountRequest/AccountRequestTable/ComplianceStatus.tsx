import React from 'react'
import { StatusCheck } from './StatusCheck'
import { IAccountRequest } from '../../../store'

// TODO define what is received
interface IComplianceStatus {
  accountRequest: IAccountRequest
}

export const ComplianceStatus: React.FC<IComplianceStatus> = (
  props: IComplianceStatus
) => {
  if (!props.accountRequest) {
    return (
      <div
        className="ni-table-compliance-status"
        data-testid="compliance-status"
      >
        Loading
      </div>
    )
  }

  const statusLabel = (check: string) => {
    switch (check) {
      case 'VALID':
        return 'Successful Verification'
      case 'INVALID':
        return 'Not valid'
      case 'MATCH':
        return 'Possible matches found'
      case 'NOMATCH':
        return 'No matches found'
      default:
        return 'Pending'
    }
  }

  const status = props.accountRequest.verificationStatus

  const faceStatus = `Face Match - ${statusLabel(status.faceStatus)}`
  const documentStatus = `Document Match - ${statusLabel(
    status.documentStatus
  )}`
  const addressStatus = `Address Check - ${statusLabel(status.addressStatus)}`
  const sanctionsStatus = `Sanctions - ${statusLabel(status.sanctionsStatus)}`
  const mediaStatus = `Adverse Media - ${statusLabel(status.mediaStatus)}`
  const politicalExposure = `Politically Exposed - ${statusLabel(
    status.politicalExposure
  )}`

  return (
    <div className="ni-table-compliance-status" data-testid="compliance-status">
      <StatusCheck title={faceStatus} verificationCheck={status.faceStatus} />
      <StatusCheck
        title={documentStatus}
        verificationCheck={status.documentStatus}
      />
      <StatusCheck
        title={addressStatus}
        verificationCheck={status.addressStatus}
      />
      <StatusCheck
        title={sanctionsStatus}
        verificationCheck={status.sanctionsStatus}
      />
      <StatusCheck title={mediaStatus} verificationCheck={status.mediaStatus} />
      <StatusCheck
        title={politicalExposure}
        verificationCheck={status.politicalExposure}
      />
    </div>
  )
}
