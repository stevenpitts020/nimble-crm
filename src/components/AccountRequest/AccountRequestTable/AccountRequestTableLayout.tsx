import React, { useContext, useState } from 'react'
import { Table, Typography } from 'antd'
import { ColumnProps } from 'antd/es/table'
import Helpers from '../../../utils/FormatHelper'
import {
  IAccountRequest,
  IAccountRequestPatch,
  ISelectOptions,
  IUser,
} from '../../../store'
import { Box, ModalWindow } from '../../index'
import { ComplianceStatus } from './ComplianceStatus'
import { ISearchOptions } from '../../../store'
import { PER_PAGE } from './constants'
import { AccountRequestStatus } from './AccountRequestStatus'
import { TablePaginationConfig } from 'antd/lib/table'
import Select, { SingleValue } from 'react-select'
import Toggle from 'react-toggle'
import _ from 'lodash'
import { AuthContextStore } from '../../../store/AuthProvider'
import { ProfileContext } from '../../../store/ProfileProvider'
import parseHtml from 'html-react-parser'

interface ITableLayout {
  data: IAccountRequest[]
  loading?: boolean
  pagination?: ISearchOptions
  count: number
  referrers?: IUser[]
  onTableChange?: (pagination: TablePaginationConfig) => void
  onRow?: (record: IAccountRequest) => React.HTMLAttributes<HTMLElement>
  onAccountRequestChange: (accountRequest: IAccountRequestPatch) => void
}

const { Text, Title } = Typography

const SHOW_DETAILS_FOR_STATUSES = ['APPROVED', 'DECLINED']

export const AccountRequestTableLayout: React.FC<ITableLayout> = (
  props: ITableLayout
) => {
  const { isInstitutionAdmin } = useContext(AuthContextStore)
  const { profileState } = useContext(ProfileContext)
  const [presentStatusDetails, setPresentStatusDetails] = useState<boolean>(
    false
  )
  const [statusDetailsTitle, setStatusDetailsTitle] = useState<string>('')
  const [statusDetailsBody, setStatusDetailsBody] = useState<string>('')

  const statusDetailsAvailable = (accountRequest: IAccountRequest) =>
    SHOW_DETAILS_FOR_STATUSES.includes(accountRequest.status)

  const hideStatusDetails = () => {
    setStatusDetailsTitle('')
    setStatusDetailsBody('')
    setPresentStatusDetails(false)
  }

  const showStatusDetails = (acctReq: IAccountRequest) => {
    if (!SHOW_DETAILS_FOR_STATUSES.includes(acctReq.status)) return
    if (!acctReq.statusEmailBody && !acctReq.statusEmailSubject) return

    setStatusDetailsTitle(acctReq.statusEmailSubject || '')

    setStatusDetailsBody(
      _.join(
        _.reduce(
          [
            { pattern: / {2,2}/g, replacement: _.repeat('&nbsp;', 2) },
            { pattern: / {3,3}/g, replacement: _.repeat('&nbsp;', 3) },
            { pattern: / {4,4}/g, replacement: _.repeat('&nbsp;', 4) },
            { pattern: / {5,5}/g, replacement: _.repeat('&nbsp;', 5) },
            { pattern: / {6,6}/g, replacement: _.repeat('&nbsp;', 6) },
            { pattern: / {7,7}/g, replacement: _.repeat('&nbsp;', 7) },
            { pattern: / {8,8}/g, replacement: _.repeat('&nbsp;', 8) },
            { pattern: / {9,9}/g, replacement: _.repeat('&nbsp;', 9) },
            { pattern: / {10,10}/g, replacement: _.repeat('&nbsp;', 10) },
            { pattern: / {11,}/g, replacement: _.repeat('&nbsp;', 10) }, // reduce 11+ literal spaces to 10 nbsp
            { pattern: /\t/g, replacement: _.repeat('&nbsp;', 4) }, // replace tabs with 4 nbsp
            { pattern: /\n/g, replacement: '<br>' }, // replace new lines with br
          ],
          (html, args) => _.replace(html, args.pattern, args.replacement), // perform the replacement
          acctReq.statusEmailBody || '' // the raw body
        ),
        '' // no join separator, just concat
      )
    )

    setPresentStatusDetails(true)
  }

  const onReferredByChange = (id: string) => (
    option: SingleValue<ISelectOptions>
  ) => {
    props.onAccountRequestChange({ id, referredById: option?.value || '!' })
  }

  const onDeleteChange = (id: string) => (event: any) => {
    if (!_.isNil(event?.target?.checked)) {
      props.onAccountRequestChange({
        id,
        deleted: !!event.target.checked,
      })
    }
  }

  const columns: ColumnProps<IAccountRequest>[] = [
    {
      title: 'Name',
      dataIndex: 'name',
      onCell: props.onRow,
      render: (text, record) => (
        <Box textAlign="left">
          <Title level={5} data-testid={`status-${record.status}`}>
            {Helpers.firstSignerFullName(record.signers)}
          </Title>
        </Box>
      ),
    },
    {
      title: 'Request Date',
      dataIndex: 'since',
      align: 'center',
      onCell: props.onRow,
      render: (text, record) => (
        <Text className="ni-color-dark">
          {Helpers.dateTimeFormatExtended(record.createdAt)}
        </Text>
      ),
    },
    {
      title: 'Referred By',
      dataIndex: 'referred-by',
      align: 'left',
      render: (text, accountRequest) => (
        <Select
          name="referred-by"
          className="referred-by"
          data-testid={`referred-by-${accountRequest.id}`}
          isMulti={false}
          isClearable={true}
          options={props.referrers?.map(ref => {
            return { value: ref.id, label: `${ref.firstName} ${ref.lastName}` }
          })}
          defaultValue={
            accountRequest.referredBy
              ? {
                  value: accountRequest?.referredBy?.id,
                  label: `${accountRequest?.referredBy?.firstName} ${accountRequest?.referredBy?.lastName}`,
                }
              : undefined
          }
          onChange={onReferredByChange(accountRequest.id)}
        />
      ),
    },
    {
      title: 'Products',
      dataIndex: 'products',
      align: 'center',
      onCell: props.onRow,
      render: (text, record) => {
        const [productConfig] = record.productConfigurations
        const productName = productConfig?.product?.name || '-'
        return <Text>{productName}</Text>
      },
    },
    {
      title: 'Status',
      dataIndex: 'status',
      align: 'center',
      render: (text, record) => (
        <AccountRequestStatus
          accountRequest={record}
          isStatusClickable={statusDetailsAvailable}
          onStatusClick={showStatusDetails}
        />
      ),
    },
    {
      title: 'Compliance Checks',
      dataIndex: 'identityVerified',
      align: 'center',
      ellipsis: true,
      onCell: props.onRow,
      render: (text, record) => <ComplianceStatus accountRequest={record} />,
    },
  ]

  if (isInstitutionAdmin(profileState?.profile))
    columns.push({
      title: 'Delete',
      dataIndex: 'delete',
      align: 'right',
      render: (text, record) => (
        <Toggle
          id="account-request-delete"
          defaultChecked={false}
          onChange={onDeleteChange(record.id)}
        />
      ),
    })

  return (
    <>
      <Table
        className={'row-click'}
        rowKey="id"
        columns={columns}
        dataSource={props.data}
        loading={props.loading}
        pagination={{
          current:
            Math.floor(
              (props.pagination?.offset || 0) /
                (props.pagination?.limit || PER_PAGE)
            ) + 1,
          pageSize: props.pagination?.limit || PER_PAGE,
          total: props.count,
          showTotal: (total, range) => (
            <span
              data-testid="pagination-total"
              data-pagination-total={range[1]}
            >
              Showing {range[0]} - {range[1]}
              {total < (props.pagination?.limit || PER_PAGE)
                ? ''
                : ` of ${total}`}{' '}
              items
            </span>
          ),
        }}
        onChange={props.onTableChange}
      />
      <ModalWindow
        closable={true}
        visible={presentStatusDetails}
        afterClose={hideStatusDetails}
        onCancel={hideStatusDetails}
        introductionTitle={
          statusDetailsTitle.length <= 150 ? statusDetailsTitle : ''
        }
        introductionText={
          statusDetailsTitle.length > 150 ? statusDetailsTitle : ''
        }
      >
        <Box className="status-details-wrapper">
          {parseHtml(statusDetailsBody)}
        </Box>
      </ModalWindow>
    </>
  )
}

AccountRequestTableLayout.defaultProps = {
  loading: false,
  data: [],
}
