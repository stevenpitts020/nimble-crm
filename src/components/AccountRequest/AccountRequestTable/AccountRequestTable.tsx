import _ from 'lodash'
import React, { useContext } from 'react'
import { useHistory } from 'react-router-dom'
import { Card, Skeleton } from 'antd'
import { AccountRequestTableLayout } from './AccountRequestTableLayout'
import {
  AccountRequestContext,
  initialSearchOptions,
} from '../../../store/AccountRequestProvider'
import { IAccountRequest, IAccountRequestPatch } from '../../../store'

import { log, accountRequestService } from '../../../services'
import { PER_PAGE } from './constants'
import { EmptyAccountRequest } from './EmptyAccountRequest'
import { TablePaginationConfig } from 'antd/lib/table'
import { ProfilesContext } from '../../../store/ProfilesProvider'

export const AccountRequestTable: React.FC = (props: any) => {
  const history = useHistory()
  const {
    accountState,
    listAccountRequests,
    currentStatusFilter,
    changeStatusFilter,
  } = useContext(AccountRequestContext)

  const { profilesState } = useContext(ProfilesContext)

  const isLoading = accountState.status === 'loading'

  React.useEffect(() => {
    const queryStr = new URLSearchParams(history.location.search)
    const limit = parseInt(queryStr.get('limit') || PER_PAGE.toString(), 10)
    const offset = parseInt(queryStr.get('offset') || '0', 10)
    const status: string | null = queryStr.get('status')

    changeStatusFilter(status)

    listAccountRequests({
      ...initialSearchOptions,
      limit,
      offset,
      status,
    })

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [history.location.search, currentStatusFilter])

  const handleTableChange = (pagination: TablePaginationConfig) => {
    log.info(pagination, 'handleTableChange')

    const limit = pagination.pageSize || PER_PAGE
    const offset = limit * ((pagination.current || 1) - 1)
    const status: string | null = currentStatusFilter
      ? `&status=${currentStatusFilter}`
      : ''

    history.push(
      `${history.location.pathname}?limit=${limit}&offset=${offset}${status}`
    )
  }

  const handleRowOnClick = (
    record: IAccountRequest
  ): React.HTMLAttributes<HTMLElement> => {
    return {
      onClick: () => {
        history.push('/accounts/' + record.id)
      }, // click row
    }
  }

  const patchAccountRequest = (accountRequest: IAccountRequestPatch) => {
    accountRequestService.patchAccountRequest(accountRequest.id, {
      ..._.omit(accountRequest, 'id'),
    })
  }

  if (isLoading) {
    return (
      <Card bordered={false} size="default">
        <Skeleton active={true} />
      </Card>
    )
  }

  return (
    <>
      {accountState.accountRequests.length > 0 && (
        <Card bordered={false} size="small">
          <AccountRequestTableLayout
            data={accountState.accountRequests}
            loading={isLoading}
            pagination={accountState.pagination}
            count={accountState.count}
            onTableChange={handleTableChange}
            onRow={handleRowOnClick}
            onAccountRequestChange={patchAccountRequest}
            referrers={
              profilesState && profilesState.profiles
                ? profilesState.profiles
                : []
            }
          />
        </Card>
      )}

      {accountState.accountRequests.length === 0 && !isLoading && (
        <EmptyAccountRequest />
      )}
    </>
  )
}
