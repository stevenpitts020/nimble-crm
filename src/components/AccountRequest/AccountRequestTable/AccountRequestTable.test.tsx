import React from 'react'
import { AccountRequestTable } from './AccountRequestTable'
import { AccountRequestProvider } from '../../../store/AccountRequestProvider'
import { waitFor } from '@testing-library/react'
import nock from 'nock'
import axios from 'axios'
import * as mocks from '../../../services/__mocks__/AccountRequest'
import { renderWithReactRouter } from '../AccountHeader/AccountHeader.test'
import Config from '../../../services/Config'

describe('AccountRequestTable', () => {
  // tslint:disable-next-line: no-var-requires
  axios.defaults.adapter = require('axios/lib/adapters/http')

  beforeEach(() => {
    nock.disableNetConnect()
    nock.enableNetConnect(/^(127\.0\.0\.1|localhost)/)
  })

  afterEach(() => {
    nock.cleanAll()
  })

  // rendering variations on env require --updateSnapshot to be run on test runner instance
  test.skip('it renders the account requests', async () => {
    const mockedResponse = [mocks.successResponseApproved]
    const mockedCountResponse = mocks.successCountResponse
    const scope = nock(`${Config.coreAPI}`)
      .defaultReplyHeaders({ 'access-control-allow-origin': '*' })
      .get(/account-requests/)
      .reply(200, mockedResponse)

    const countScope = nock(`${Config.coreAPI}`)
      .defaultReplyHeaders({ 'access-control-allow-origin': '*' })
      .get(/account-requests\/count/)
      .reply(200, mockedCountResponse)

    const { container } = renderWithReactRouter(
      <AccountRequestProvider>
        <AccountRequestTable />
      </AccountRequestProvider>
    )

    await waitFor(() => {
      scope.done()
      countScope.done()
    })
    expect(container).toMatchSnapshot()
  })

  test('it renders empty component', async () => {
    const mockedResponse = [] as any
    const scope = nock(`${Config.coreAPI}`)
      .defaultReplyHeaders({ 'access-control-allow-origin': '*' })
      .get(/account-requests/)
      .reply(200, mockedResponse)

    const { getByTestId } = renderWithReactRouter(
      <AccountRequestProvider>
        <AccountRequestTable />
      </AccountRequestProvider>
    )

    await waitFor(() => {
      const empty = getByTestId('empty-accounts')
      expect(empty).toBeInTheDocument()
    })
    scope.done()
  })
})
