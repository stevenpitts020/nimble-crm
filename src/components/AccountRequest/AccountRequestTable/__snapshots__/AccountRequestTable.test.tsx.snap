// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountRequestTable it renders the account requests 1`] = `
<div>
  <div
    class="ant-card ant-card-small"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-table-wrapper row-click"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: fixed;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          Name
                        </th>
                        <th
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          Request Date
                        </th>
                        <th
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          Products
                        </th>
                        <th
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          Status
                        </th>
                        <th
                          class="ant-table-cell ant-table-cell-ellipsis"
                          style="text-align: center;"
                          title="Compliance Checks"
                        >
                          Compliance Checks
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="2552ab85-08da-4bb5-be00-9e94d282d348"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ni-box u-align-left"
                            data-testid="box"
                          >
                            <h5
                              class="ant-typography"
                              data-testid="status-APPROVED"
                            >
                              Han James Solo
                            </h5>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <span
                            class="ant-typography ni-color-dark"
                          >
                            Mar 16, 2019 12:34 PM
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <span
                            class="ant-typography"
                          >
                            Simple Checking
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <div
                            class="account-request-status"
                          >
                            <button
                              class="approved"
                            >
                              Approved
                            </button>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell ant-table-cell-ellipsis"
                          style="text-align: center;"
                        >
                          <div
                            class="ni-table-compliance-status"
                            data-testid="compliance-status"
                          >
                            <div
                              class="status-check-item"
                            >
                              <span
                                aria-hidden="true"
                                hidden=""
                              >
                                Face Match - Successful Verification
                              </span>
                              <span
                                class="anticon"
                                data-testid="passed"
                                role="img"
                                style="font-size: 24px;"
                              >
                                <svg
                                  fill="none"
                                  height="1em"
                                  viewBox="0 0 24 24"
                                  width="1em"
                                >
                                  <path
                                    clip-rule="evenodd"
                                    d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
                                    fill="#36B37E"
                                    fill-rule="evenodd"
                                    opacity=".15"
                                  />
                                  <path
                                    d="M10.398 15.824l-3.261-3.242A.476.476 0 017 12.23c0-.143.046-.26.137-.351l.722-.703a.435.435 0 01.342-.156c.137 0 .257.052.361.156l2.188 2.187 4.688-4.687a.496.496 0 01.36-.156c.137 0 .251.052.343.156l.722.703A.476.476 0 0117 9.73c0 .144-.046.26-.137.352l-5.761 5.742a.446.446 0 01-.352.156.446.446 0 01-.352-.156z"
                                    fill="#01C48D"
                                  />
                                </svg>
                              </span>
                            </div>
                            <div
                              class="status-check-item"
                            >
                              <span
                                aria-hidden="true"
                                hidden=""
                              >
                                Document Match - Not valid
                              </span>
                              <span
                                class="anticon"
                                data-testid="failed"
                                role="img"
                                style="font-size: 24px;"
                              >
                                <svg
                                  fill="none"
                                  height="1em"
                                  viewBox="0 0 24 24"
                                  width="1em"
                                >
                                  <path
                                    clip-rule="evenodd"
                                    d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
                                    fill="#F97A7A"
                                    fill-rule="evenodd"
                                  />
                                  <path
                                    clip-rule="evenodd"
                                    d="M12 6a1 1 0 00-1 1v6a1 1 0 102 0V7a1 1 0 00-1-1zm0 10a1 1 0 110 2 1 1 0 010-2z"
                                    fill="#FCFDFF"
                                    fill-rule="evenodd"
                                  />
                                </svg>
                              </span>
                            </div>
                            <div
                              class="status-check-item"
                            >
                              <span
                                aria-hidden="true"
                                hidden=""
                              >
                                Address Check - Not valid
                              </span>
                              <span
                                class="anticon"
                                data-testid="failed"
                                role="img"
                                style="font-size: 24px;"
                              >
                                <svg
                                  fill="none"
                                  height="1em"
                                  viewBox="0 0 24 24"
                                  width="1em"
                                >
                                  <path
                                    clip-rule="evenodd"
                                    d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
                                    fill="#F97A7A"
                                    fill-rule="evenodd"
                                  />
                                  <path
                                    clip-rule="evenodd"
                                    d="M12 6a1 1 0 00-1 1v6a1 1 0 102 0V7a1 1 0 00-1-1zm0 10a1 1 0 110 2 1 1 0 010-2z"
                                    fill="#FCFDFF"
                                    fill-rule="evenodd"
                                  />
                                </svg>
                              </span>
                            </div>
                            <div
                              class="status-check-item"
                            >
                              <span
                                aria-hidden="true"
                                hidden=""
                              >
                                Sanctions - No matches found
                              </span>
                              <span
                                class="anticon"
                                data-testid="passed"
                                role="img"
                                style="font-size: 24px;"
                              >
                                <svg
                                  fill="none"
                                  height="1em"
                                  viewBox="0 0 24 24"
                                  width="1em"
                                >
                                  <path
                                    clip-rule="evenodd"
                                    d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
                                    fill="#36B37E"
                                    fill-rule="evenodd"
                                    opacity=".15"
                                  />
                                  <path
                                    d="M10.398 15.824l-3.261-3.242A.476.476 0 017 12.23c0-.143.046-.26.137-.351l.722-.703a.435.435 0 01.342-.156c.137 0 .257.052.361.156l2.188 2.187 4.688-4.687a.496.496 0 01.36-.156c.137 0 .251.052.343.156l.722.703A.476.476 0 0117 9.73c0 .144-.046.26-.137.352l-5.761 5.742a.446.446 0 01-.352.156.446.446 0 01-.352-.156z"
                                    fill="#01C48D"
                                  />
                                </svg>
                              </span>
                            </div>
                            <div
                              class="status-check-item"
                            >
                              <span
                                aria-hidden="true"
                                hidden=""
                              >
                                Adverse Media - No matches found
                              </span>
                              <span
                                class="anticon"
                                data-testid="passed"
                                role="img"
                                style="font-size: 24px;"
                              >
                                <svg
                                  fill="none"
                                  height="1em"
                                  viewBox="0 0 24 24"
                                  width="1em"
                                >
                                  <path
                                    clip-rule="evenodd"
                                    d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
                                    fill="#36B37E"
                                    fill-rule="evenodd"
                                    opacity=".15"
                                  />
                                  <path
                                    d="M10.398 15.824l-3.261-3.242A.476.476 0 017 12.23c0-.143.046-.26.137-.351l.722-.703a.435.435 0 01.342-.156c.137 0 .257.052.361.156l2.188 2.187 4.688-4.687a.496.496 0 01.36-.156c.137 0 .251.052.343.156l.722.703A.476.476 0 0117 9.73c0 .144-.046.26-.137.352l-5.761 5.742a.446.446 0 01-.352.156.446.446 0 01-.352-.156z"
                                    fill="#01C48D"
                                  />
                                </svg>
                              </span>
                            </div>
                            <div
                              class="status-check-item"
                            >
                              <span
                                aria-hidden="true"
                                hidden=""
                              >
                                Politically Exposed - Possible matches found
                              </span>
                              <span
                                class="anticon"
                                data-testid="failed"
                                role="img"
                                style="font-size: 24px;"
                              >
                                <svg
                                  fill="none"
                                  height="1em"
                                  viewBox="0 0 24 24"
                                  width="1em"
                                >
                                  <path
                                    clip-rule="evenodd"
                                    d="M12 24c6.627 0 12-5.373 12-12S18.627 0 12 0 0 5.373 0 12s5.373 12 12 12z"
                                    fill="#F97A7A"
                                    fill-rule="evenodd"
                                  />
                                  <path
                                    clip-rule="evenodd"
                                    d="M12 6a1 1 0 00-1 1v6a1 1 0 102 0V7a1 1 0 00-1-1zm0 10a1 1 0 110 2 1 1 0 010-2z"
                                    fill="#FCFDFF"
                                    fill-rule="evenodd"
                                  />
                                </svg>
                              </span>
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
            <ul
              class="ant-pagination ant-table-pagination ant-table-pagination-right"
              unselectable="unselectable"
            >
              <li
                class="ant-pagination-total-text"
              >
                <span
                  data-pagination-total="2"
                  data-testid="pagination-total"
                >
                  Showing
                  1
                   -
                  2


                  items
                </span>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-prev ant-pagination-disabled"
                title="Previous Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="left"
                    class="anticon anticon-left"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a
                  rel="nofollow"
                >
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-next ant-pagination-disabled"
                title="Next Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="right"
                    class="anticon anticon-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
